WalletWasabi.Packager/MacSignTools.cs

```diff
@@ -0,0 +1,394 @@
+			var contentsPath = Path.GetFullPath(Path.Combine(Program.PackagerProjectDirectory.Replace("\\", "//"), "Content", "Osx"));
+			var entitlementsPath = Path.Combine(contentsPath, "entitlements.plist");
+			var entitlementsSandBoxPath = Path.Combine(contentsPath, "entitlements_sandbox.plist");
+			var torZipDirPath = Path.Combine(binPath, "TorDaemons", "tor-osx64");
+			var torZipPath = $"{torZipDirPath}.zip";
+			var desktopDmgFilePath = Path.Combine(desktopPath, dmgFileName);
+
+			var signArguments = $"--sign \"L233B2JQ68\" --verbose --options runtime --timestamp";
+
+			Console.WriteLine("Phase: creating the working directory.");
+
+			Console.WriteLine("Enter appleId (email):");
+			var appleId = Console.ReadLine();
+			Console.WriteLine("Enter password:");
+			var password = Console.ReadLine();
+
+			if (Directory.Exists(workingDir))
+			{
+				DeleteWithChmod(workingDir);
+			}
+
+			if (File.Exists(desktopDmgFilePath))
+			{
+				File.Delete(desktopDmgFilePath);
+			}
+
+			Console.WriteLine("Phase: creating the app.");
+
+			IoHelpers.EnsureDirectoryExists(binPath);
+			ZipFile.ExtractToDirectory(zipPath, binPath); // Copy the binaries.
+
+			IoHelpers.CopyFilesRecursively(new DirectoryInfo(Path.Combine(contentsPath, "App")), new DirectoryInfo(appPath));
+
+			Console.WriteLine("Update the plist file with current information for example with version.");
+
+			var lines = File.ReadAllLines(infoFilePath);
+			string bundleIdentifier = null;
+
+			for (int i = 0; i < lines.Length; i++)
+			{
+				string line = lines[i];
+				if (!line.TrimStart().StartsWith("<key>", StringComparison.InvariantCultureIgnoreCase))
+				{
+					continue;
+				}
+
+				if (line.Contains("CFBundleShortVersionString", StringComparison.InvariantCulture) ||
+					line.Contains("CFBundleVersion", StringComparison.InvariantCulture))
+				{
+					lines[i + 1] = lines[i + 1].Replace("?", $"{Version.Parse(versionPrefix).ToString(3)}"); // Apple allow only 3 version tags in plist.
+				}
+				else if (line.Contains("CFBundleIdentifier", StringComparison.InvariantCulture))
+				{
+					bundleIdentifier = lines[i + 1].Trim().Replace("<string>", "").Replace("</string>", "");
+				}
+			}
+			IoHelpers.DeleteRecursivelyWithMagicDustAsync(infoFilePath).GetAwaiter().GetResult();
+			File.WriteAllLines(infoFilePath, lines);
+
+			using (var process = Process.Start(new ProcessStartInfo
+			{
+				FileName = "chmod",
+				Arguments = $"-R u+rwX,go+rX,go-w \"{appPath}\"",
+				WorkingDirectory = workingDir
+			}))
+			{
+				process.WaitForExit();
+			}
+
+			var filesToCheck = new[] { entitlementsPath, entitlementsSandBoxPath, torZipPath };
+
+			foreach (var file in filesToCheck)
+			{
+				if (!File.Exists(file))
+				{
+					throw new FileNotFoundException($"File missing: {file}");
+				}
+			}
+
+			// Tor already signed by: The Tor Project, Inc (MADPSAYN6T)
+
+			var exacutableFileNames = new[] { "wassabee","bitcoind","hwi"};
```

Comment:
What about Tor? Is it different?

Reply:
It is already signed by The Tor Project, Inc (MADPSAYN6T). So IMO we do not need to sign it again.